service/topology_coordinator.cc

```diff
@@ -740,6 +755,82 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         }
     }
 
+    // If a node crashes after initiating gossip and before joining group0, it becomes orphan and 
+    // remains in gossip. This backgound fiber periodically checks for such nodes and purges those 
+    // entries from gossiper by commiting the node as left in raft.
+    future<> gossiper_orphan_remover_fiber() {
+        rtlogger.debug("start gossiper orphan remover fiber");
+        bool do_speedup_fiber = false;
+        co_await utils::get_local_injector().inject("fast_orphan_removal_fiber", [&](auto& handler) -> future<> {
+            do_speedup_fiber = true;
+            co_return;
+        });
+        while (!_as.abort_requested()) {
+            try {
+                auto guard = co_await start_operation();
+                std::vector<canonical_mutation> updates;
+                std::optional<gms::inet_address> orphan_ip;
+                co_await utils::get_local_injector().inject("speedup_orphan_removal", [&](auto& handler) -> future<> {
+                    auto fetched_orphan_ip = handler.get("orphan_ip");
+                    orphan_ip = gms::inet_address(sstring{*fetched_orphan_ip});
```

Comment:
ok, I see, makes sense

Reply:
>If you change the timeout through an error injection after all initial nodes join the cluster, the test shouldn't become flaky

Right, but again, you will have to adjust the timeout to some value other than 60secs. Even 10secs or 5 secs seem problematic to me, since we wait for some hardcoded time. I dont think making it 0  is a good idea, since it would be bit absurd.

btw orphan_ip is actually orphan(we know that in test) , and we are just fast-tracking its removal via injection.
I am unsure if making timeout 0  is a good idea.
@kbr-scylla thoughts?

Reply:
>  @kbr-scylla thoughts?

Dunno. I'll let you guys figure it out.

Reply:
I made this change. Since we are using the injection for this specific test, it allows us to "remove all unjoined nodes", and still be correct. I have also commented about the same.

This way readability is also better!!